examples/evm/its-lock-unlock-fee/FeeToken.sol

```diff
@@ -0,0 +1,124 @@
+
+    uint256 internal constant UINT256_MAX = 2 ** 256 - 1;
+
+    uint256 public feePercent;
+
+    /**
+     * @notice Constructs the InterchainToken contract.
+     * @dev Makes the implementation act as if it has been setup already to disallow calls to init() (even though that would not achieve anything really).
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimalsValue,
+        uint256 _feePercent,
+        address _interchainTokenServiceAddress
+    ) ERC20(_name, _symbol) {
+        decimals_ = _decimalsValue;
+        interchainTokenService_ = _interchainTokenServiceAddress;
+
+        feePercent = _feePercent;
+
+        _addMinter(msg.sender);
+    }
+
+    function decimals() public view override returns (uint8) {
+        return decimals_;
+    }
+
+    /**
+     * @notice set new fee percent
+     * @param _feePercent new fee percent
+     */
+    function setFeePercent(uint256 _feePercent) external onlyRole(uint8(Roles.MINTER)) {
+        require(_feePercent <= 100, 'Fee percent too high');
+        feePercent = _feePercent;
+    }
+
+    /**
+     * @notice Returns the interchain token service
+     * @return address The interchain token service contract
+     */
+    function interchainTokenService() public view override returns (address) {
+        return interchainTokenService_;
+    }
+
+    /**
+     * @notice Returns the tokenId for this token.
+     * @return bytes32 The token manager contract.
+     */
+    function interchainTokenId() public view override returns (bytes32) {
+        return tokenId;
+    }
+
+    /**
+     * @notice Function to mint new tokens.
+     * @dev Can only be called by the minter address.
+     * @param account The address that will receive the minted tokens.
+     * @param amount The amount of tokens to mint.
+     */
+    function mint(address account, uint256 amount) external onlyRole(uint8(Roles.MINTER)) {
+        _mint(account, amount);
+    }
+
+    /**
+     * @notice Function to burn tokens.
+     * @dev Can only be called by the minter address.
+     * @param account The address that will have its tokens burnt.
+     * @param amount The amount of tokens to burn.
+     */
+    function burn(address account, uint256 amount) external onlyRole(uint8(Roles.MINTER)) {
+        _burn(account, amount);
+    }
+
+    function addMinter(address minter) external {
+        _addMinter(minter);
+    }
+
+    /**
+     * @notice transfer token from sender to recipient. Sender must approve tx to be sent
+     * @param from sender of the token
+     * @param to recipient of the token
+     * @param amount amount of tokens to be sent
+     */
+    function transferFrom(address from, address to, uint256 amount) public override returns (bool) {
+        uint256 fee = (amount * feePercent) / 1e18;
+        uint256 amountAfterFee = amount - fee;
+
+        super.transferFrom(from, to, amountAfterFee);
+
+        _burn(from, fee);
```

Comment:
Fees are burned, what if they want to collect the fees?

Reply:
Ya the `burn()` is certainly not a requirement, the point is to show how this fee is deducted from what is sent, to what arrives at the receiver's address. We could send it to a community pool or even do nothing with it, I did not do a community pool here to handle the fee just out of the interested of keeping this simple, I burned the fee just to update the supply for tokens that are not really being handled, figured it was just easiest to burn it.

Reply:
Can you add a comment here that says you can do whatever you want with the transaction fees (give a couple examples)